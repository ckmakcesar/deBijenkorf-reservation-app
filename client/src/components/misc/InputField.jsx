import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({
  label,
  name,
  value,
  type,
  min,
  max,
  onChange,
  required,
  disabled,
}) => {
  const optionalProps = {};
  if (name) {
    optionalProps.name = name;
  }
  if (type) {
    optionalProps.type = type;
    if (type === 'number' && !isNaN(min)) {
      optionalProps.min = min;
    }
    if (type === 'number' && !isNaN(max)) {
      optionalProps.max = max;
    }
  }

  return (
    <label className='input-field-label'>
      <div>{label ? `${label}${label && required ? '*' : ''}` : undefined}</div>

      <input
        value={value}
        onChange={e => onChange(e.target.value)}
        required={required}
        disabled={disabled}
        {...optionalProps}
      />
    </label>
  );
};

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
InputField.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  min: PropTypes.number,
  max: PropTypes.number,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
};

InputField.defaultProps = {
  label: '',
  name: '',
  value: '',
  type: '',
  min: NaN,
  max: NaN,
  required: false,
  disabled: false,
};

export default InputField;
