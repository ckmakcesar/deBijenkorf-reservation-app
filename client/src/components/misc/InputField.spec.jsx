import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputField from './InputField';

describe('component:InputField', () => {
  it('Should render InputField with every optional props absent', () => {
    const wrapper = shallow(
      <InputField
        onChange={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render InputField with every optional props present', () => {
    const wrapper = shallow(
      <InputField
        label='Number Input'
        name='number-input'
        value={(555).toString()}
        type='number'
        min={100}
        max={9999}
        onChange={jest.fn()}
        required={true}
        disabled={false}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
