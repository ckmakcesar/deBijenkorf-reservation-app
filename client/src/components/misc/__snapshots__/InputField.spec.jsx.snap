// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:InputField Should render InputField with every optional props absent 1`] = `
<label
  className="input-field-label"
>
  <div />
  <input
    disabled={false}
    onChange={[Function]}
    required={false}
    value=""
  />
</label>
`;

exports[`component:InputField Should render InputField with every optional props present 1`] = `
<label
  className="input-field-label"
>
  <div>
    Number Input*
  </div>
  <input
    disabled={false}
    max={9999}
    min={100}
    name="number-input"
    onChange={[Function]}
    required={true}
    type="number"
    value="555"
  />
</label>
`;
